docs/ValidationRules/RULEID.RULEFRIENDLYNAME.cs

```diff
@@ -0,0 +1,142 @@
+             *      All sentences together will be used as a `LongDescription` for this rule.
+             */
+            Text = RuleResources.RULEID_RULEFRIENDLYNAME_Description
+        };
+
+        public override MultiformatMessageString FullDescription => _fullDescription;
+
+        /*
+         * INSTRUCTIONS:
+         * Decide the appropriate FailureLevel for this rule as appropriate.
+         * The following heuristics can be used to arrive at a decision:
+         * 
+         * Error:
+         *      In general, an `Error` should be reserved for rules which address
+         *      a SARIF spec violation that cannot be expressed by the Schema.
+         *      Example:
+         *      SARIF1019.RuleIdMustBePresentAndConsistent
+         *      Per spec, at least one of result.ruleId and result.rule.id must be present, 
+         *      and if both are present, they must be the same.
+         *      
+         * Warning:
+         *      In general, a `Warning` is a good practice which should be followed, but it
+         *      does not violate the SARIF spec.
+         *      Example:
+         *      SARIF1021.DoNotUseFriendlyNameAsRuleId
+         */
+        public override FailureLevel DefaultLevel => FailureLevel.Warning;
+
+
+        /*
+         * INSTRUCTIONS:
+         * Add a new property in ~\src\Sarif.Multitool\Rules\RuleId.cs with 
+         * the name as RULEFRIENDLYNAME and 
+         * the value as RULEID.
+         * 
+         * Example:
+         *      public const string ReferToFinalSchema = "SARIF1020";
+         */
+        /// <summary>
+        /// RULEID
+        /// </summary>
+        public override string Id => RuleId.RULEFRIENDLYNAME;
+
+        protected override IEnumerable<string> MessageResourceNames => new string[]
+        {
+            /*
+            * INSTRUCTIONS:
+            * Each rule has one or more result message strings, each with symbolic name 
+            * in PascalCase. Add atleast one new key-value pair for user messages
+            * in ~\Sarif.Multitool\Rules\RuleResources.resx.
+            *
+            * Key: 
+            *       RULEID_USERMESSAGESYMBOLICNAME
+            * 
+            * Value: 
+            *      Provide the default user message for this rule. It should be a dynamic string
+            *      and always start with `{0}:`.
+            * 
+            * Conditional user messages:
+            *      If the rule requires more than one user messages to be defined, define each as a
+            *      key-value pair in resx file. The keys should be named as: RULEID_SHORTDESCRIPTION
+            * 
+            * Example:
+            * 
+            *      Key      : SARIF1018_LacksTrailingSlash
+            *      Value    : {0}: The URI '{1}' belonging to the '{2}' element of 
+            *                 run.originalUriBaseIds does not end with a slash.
+            */
```

Comment:
>* [](start = 12, length = 1)

 * NOTE: Provide a meaningful symbolic name for each message, even if there is only one in this rule. Do not use a generic name like "default" for a single message.